pyscriptjs/src/components/pyscript.ts

```diff
@@ -200,30 +121,57 @@ function createElementsWithEventListeners(interpreter: InterpreterClient, pyAttr
                 logger.error((e as Error).message);
             }
         } else {
-            el.addEventListener(event, () => {
+            el.addEventListener(browserEvent, (evt) => {
                 void (async () => {
                     try {
-                        await interpreter.run(handlerCode);
-                    } catch (err) {
+                        const pyEval = interpreter.globals.get('eval') //do i need globals here
+                        const pyCallable = interpreter.globals.get('callable')
+                        const pyDictClass = interpreter.globals.get('dict')
+
+                        const localsDict = pyDictClass()
+                        localsDict.set('event', evt)
+
+                        const evalResult = pyEval(userProvidedFunctionName, interpreter.globals, localsDict)
+                        const isCallable = pyCallable(evalResult)
+
+                        if (isCallable) {
+                            const pyInspectModule = interpreter.interface.pyimport('inspect')
```

Comment:
@antocuni this is what I need the `interface` for

Reply:
if I understand it correctly, here you are trying to detect whether the user provided something which is not a callable to show a better error message, aren't you?

If so, I propose the following plan:
1. for now, you should be able to fix it using `interpreter._remote.interface`. But note that when you use the `RemoteInterpreter` everything is async, so I suspect you will have to put `await` in front of every line. This should at least be able to unblock you

2. for later, we might want to move this logic directly to `RemoteInterpreter`: e.g., `RemoteInterpreter` could have a method called `isPyCallable()` which checks whether the given name is actually a callable, and a corresponding method on `InterpreterClient`. But I don't really understand why you need to inject the `evt` argument so I want to understand it more

Reply:
it worked without await 💃